### DISCUSSION TITLE
Waiting for MySQL to come up.....

### DISCUSSION THREAD

**Discussion by Thebigbignooby** [owid/owid-grapher#3122] [Q&A] [answered]
I followed the [instructions](https://github.com/owid/owid-grapher/blob/master/docs/docker-compose-mysql.md)

After running `make up` and waiting for 10-20 minutes, I have a running MySQL database in docker with all the data, that I am able to connect to using DBeaver.

However:

The various tmux tabs show that all is fine, except for the tab named `1:admin` which is showing
`Waiting for MySQL to come up.......`
and none of the links in the tab `4:welcome` seems to be working.

On a whim, I tried running `yarn startAdminServer` in a new terminal window and I am now able to
access the site at `localhost:3030`
login to the admin at `localhost:3030/admin`

It looks like the `1:admin` service is not detecting that the Database is ready.

**Comment by danyx23** [comment, marked as answer]
@Thebigbignooby thanks for reporting this issue! That sounds a bit odd. If you run `mysql` in a local shell, does that work? If you run `make up` now after it has worked a first time, does it bring the admin up now?

If `mysql` is not installed, can you try `brew install mysql-client` and see if that solves the issue (assuming that you are on MacOs)? You may have to restart the terminal window afterwards to make sure the changes to your PATH are applied.

**Reply by Thebigbignooby** [reply]
@danyx23 Thanks for the quick reply!

> If you run make up now after it has worked a first time, does it bring the admin up now?

No

> If you run mysql in a local shell, does that work?

I'm not sure what you mean by "local shell".
But no, I don't have any MySQL executable installed. (I'm also confused as to why this would be necessary... Am I missing something?)

-> Running `brew install mysql-client` ... and added it to my path

`make up` `1:admin` pane is still `waiting for MySQL to come up....`

And running `mysql` in a different tab produces the following:

```sh
❯ mysql
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)
```

**Reply by Thebigbignooby** [reply]
Scratch that, I hadn't refreshed my "make up" terminal.
With `mysql` accessible in the PATH, the `1:admin` pane now correctly detected the database and launched the dev server.

**Reply by danyx23** [reply]
Ah that is great to hear! We ran through the setup instructions once before sending out the challenge but that was not on a freshly set up machine. I'll add it to the instructions so that others don't get tripped up by this. Thanks again for the report!

**Reply by bigxalx** [reply]
I ran into the same issue when setting up the project, however before finding this thread, I also tried using the dev container with VSCode locally as well as GitHub Codespaces with no luck. This issue exists in these conditions too. 
Also, in addition to `yarn startAdminServer` I had to run `yarn run tsc-watch -b`, as otherwise the admin frontend is hot reloading, but the backend (apiRouter) is not. I had to restart the adminServer after each change to the apiRouter before the changes were reflected. Hope this is helpful for someone running into the same issues.
